datastore_api/api/common/dependencies.py

```diff
@@ -4,16 +4,22 @@
 from fastapi import Depends, Request
 
 from datastore_api.adapter import db
-from datastore_api.common.exceptions import DatastoreNotFoundException
+from datastore_api.common.exceptions import (
+    DatastoreNotFoundException,
+    DatastoreRdnMissingException,
+)
 
 logger = logging.getLogger()
 
 
-# TODO: Remove Optional RDN + default once legacy routers are removed.
+def get_datastore_rdn_from_request(request: Request) -> str:
```

Comment:
We don't need this function anymore as there should be no methods that are being used by both rdn-containing and not-rdn-containing paths. So the lookups to this function should instead be replaced by simple path arguments for fastapi to auto inject into 👍🏻